STL_Extension/include/CGAL/callback.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2016       GeometryFactory Sarl (France)
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org); you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of the License,
+// or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+// Author(s)     : Laurent Rineau
+//
+
+#ifndef CGAL_CALLBACK_H
+#define CGAL_CALLBACK_H
+
+#include <CGAL/config.h>
+
+#ifdef CGAL_CXX11
+#  include <functional>
+#  define CGAL_CALLBACK_PARAM(x) x
+#  define CGAL_CALLBACK(f, ...) if(f) f(__VA_ARGS__);
+#  define CGAL_CALLBACK_VAR(x) x
+#else
+#  define CGAL_CALLBACK_PARAM(x) void* = 0
+#  define CGAL_CALLBACK(f)
+#  define CGAL_CALLBACK(f, p1)
+#  define CGAL_CALLBACK(f, p1, p2)
+#  define CGAL_CALLBACK(f, p1, p2, p3)
+#  define CGAL_CALLBACK(f, p1, p2, p3, p4)
+#  define CGAL_CALLBACK(f, p1, p2, p3, p4, p5)
+#  define CGAL_CALLBACK_VAR(x) 0
```

Comment:
(In reply to https://github.com/CGAL/cgal/pull/1036#issuecomment-217857764, by @sloriot)

Oops. I forgot that C++03 not only does not allow variadic macros but it does not allow either to define a macro with several number of arguments.

Probably the lines 33-38 should define macros with different names:
```C++
#  define CGAL_CALLBACK0(f)
#  define CGAL_CALLBACK1(f, p1)
#  define CGAL_CALLBACK2(f, p1, p2)
#  define CGAL_CALLBACK3(f, p1, p2, p3)
#  define CGAL_CALLBACK4(f, p1, p2, p3, p4)
#  define CGAL_CALLBACK5(f, p1, p2, p3, p4, p5)
```
and same for the C++11 part. But the API is not as nice as the previous one.

Or I could use:
```C++
 #  define CGAL_CALLBACK(...)
```
depending on the fact that all compilers we support probably support the variadic macros.